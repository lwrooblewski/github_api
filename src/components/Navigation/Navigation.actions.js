export function getArrayOfNavigationLinks(routes) {
	const routesEntries = Object.entries(routes);
	const shouldBeDisplayedRoutes = routesEntries.filter(([, { displayInNavigation }]) => displayInNavigation);
	return shouldBeDisplayedRoutes.map(([, { url, name }]) => ({
		url,
		name,
	}));
}

export function getRootUrl(routes) {
	if (!routes.root) throw new Error('No route with `root` key in routes.js');
	return routes.root.url;
}
