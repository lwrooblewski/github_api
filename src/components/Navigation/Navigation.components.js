/* eslint import/prefer-default-export: 0 */
import styled from '@emotion/styled';

export const NavigationContainer = styled.div`
	margin-top: 20px;
	margin-bottom: 20px;
`;
