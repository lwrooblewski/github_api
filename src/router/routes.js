import { Home, UsersSearch, UsersSingle, UsersList } from '../pages';

const routes = {
	root: {
		name: 'Home',
		url: '/',
		component: Home,
		displayInNavigation: true,
	},
	usersList: {
		name: 'GitHub users',
		url: '/users',
		component: UsersList,
		displayInNavigation: true,
	},
	searchUser: {
		name: 'Search for user',
		url: '/search',
		component: UsersSearch,
		displayInNavigation: true,
	},
	singleUser: {
		name: 'Single user',
		url: '/single/:username',
		component: UsersSingle,
		displayInNavigation: false,
	},
};

export default routes;
