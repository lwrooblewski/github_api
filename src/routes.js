const routes = {
  root: {
    name: 'Home',
    url: '/',
  },
  usersList: {
    name: 'GitHub users',
    url: '/users',
  },
  searchUser: {
    name: 'Search for user',
    url: '/search',
  },
};

export default routes;
