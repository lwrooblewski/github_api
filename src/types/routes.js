import { objectOf } from 'prop-types';
import route from './route';

const routes = objectOf(route).isRequired;

export default routes;
