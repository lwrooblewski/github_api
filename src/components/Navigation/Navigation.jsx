import React from 'react';
import { Nav, NavItem, NavLink } from 'react-bootstrap';
import { NavigationContainer } from './Navigation.components';
import { routesType } from '../../types';

function Navigation({ routes }) {
	const ROOT_URL = routes.root.url;
	const ROUTES_ENTRIES = Object.entries(routes);
	return (
		<NavigationContainer>
			<Nav fill variant="tabs" defaultActiveKey={ROOT_URL}>
				{ROUTES_ENTRIES.map(([key, { url, name }]) => (
					<NavItem key={key}>
						<NavLink href={url}>{name}</NavLink>
					</NavItem>
				))}
			</Nav>
		</NavigationContainer>
	);
}

Navigation.propTypes = {
	routes: routesType.isRequired,
};

export default Navigation;
