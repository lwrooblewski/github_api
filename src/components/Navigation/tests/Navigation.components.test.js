import React from 'react';
import renderer from 'react-test-renderer';
import {
	BrowserRouter,
} from "react-router-dom";
import Navigation from '../Navigation';
import routes from '../../../router/routes';

it('renders correctly', () => {
	const tree = renderer
		.create(
			<BrowserRouter>
				<Navigation routes={routes} />
			</BrowserRouter>,
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
