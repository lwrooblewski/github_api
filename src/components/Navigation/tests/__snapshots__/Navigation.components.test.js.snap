// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.emotion-0 {
  margin-top: 100px;
}

.emotion-0 {
  margin-top: 100px;
}

<div
  className="emotion-0"
>
  <div
    className="nav nav-tabs nav-fill"
    onKeyDown={[Function]}
  >
    <div
      className="nav-item"
    >
      <a
        className="nav-link active"
        data-rb-event-key="/"
        href="/"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Home
      </a>
    </div>
    <div
      className="nav-item"
    >
      <a
        className="nav-link"
        data-rb-event-key="/users"
        href="/users"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        GitHub users
      </a>
    </div>
    <div
      className="nav-item"
    >
      <a
        className="nav-link"
        data-rb-event-key="/search"
        href="/search"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Search for user
      </a>
    </div>
  </div>
</div>
`;
